usecases/config/runtime/feature_flag.go

```diff
@@ -0,0 +1,210 @@
+					for _, onChange := range f.onChange {
+						onChange(old, v)
+					}
+				}
+			}
+		}
+	}()
+}
+
+// Close ensure that all internal ressources are freed
+func (f *FeatureFlag[T]) Close() {
+	if f.ldInteg != nil {
+		f.ldInteg.ldClient.GetFlagTracker().RemoveFlagValueChangeListener(f.ldChangeChannel)
+	}
+}
+
+// WithEnvDefault configures the feature flag to read a default value from envVariable.
+// If envVariable is not present in the env the default value will not change
+// If envVariable returns a value that is not convertible to type T the value will not change.
+// Type string is directly casted
+// Type int is parsed with strconv.Atoi
+// Type float64 is parsed with strconv.ParseFloat
+// Type bool is parsed with strconv.ParseBool
+func (f *FeatureFlag[T]) WithEnvDefault(envVariable string) *FeatureFlag[T] {
+	val, ok := os.LookupEnv(envVariable)
+	// Return early if the env variable is absent
+	if !ok {
+		return f
+	}
+
+	// We're going to both read and write the value
+	f.mu.Lock()
+	defer f.mu.Unlock()
+	// Parse val into the right type expected by T
+	switch any(f.value).(type) {
+	case string:
+		if v, ok := any(val).(T); !ok {
+			f.logger.Warn("could not cast string from env, ignoring env value")
+		} else {
+			f.value = v
+			f.logger.Debug("parsed string from env, value is %s", f.value)
+		}
+	case int:
+		if v, err := strconv.Atoi(val); err != nil {
+			f.logger.Warn("could not parse int from env, ignoring env value: %w", err)
+		} else {
+			f.value = any(v).(T)
+			f.logger.Debug("parsed int from env, value is %d", f.value)
+		}
+	case float64:
+		if v, err := strconv.ParseFloat(val, 64); err != nil {
+			f.logger.Warn("could not parse float64 from env, ignoring env value: %w", err)
+		} else {
+			f.value = any(v).(T)
+			f.logger.Debug("parsed float64 from env, value is %f", f.value)
+		}
+	case bool:
+		if v, err := strconv.ParseBool(val); err != nil {
+			f.logger.Warn("could not parse bool from env, ignoring env value: %w", err)
+		} else {
+			f.value = any(v).(T)
+			f.logger.Debug("parsed bool from env, value is %t", f.value)
+		}
+	default:
+		// This should not happen
+		f.logger.Warnf("unsuported feature flag value type, got %T", f.value)
+	}
+	return f
+}
+
+// WithOnChangeCallback registers onChange to be called everytime the value of the feature flag is changed at runtime
+// These callbacks are blocking in the loop to update the feature flag, therefore the callback *must not* block.
+func (f *FeatureFlag[T]) WithOnChangeCallback(onChange func(T, T)) *FeatureFlag[T] {
+	f.onChange = append(f.onChange, onChange)
+	return f
+}
+
+// WithLogOnChange configures featureFlag to log with Info level everytime a change in the feature flag is detected.
+func (f *FeatureFlag[T]) WithLogOnChange() *FeatureFlag[T] {
+	return f.WithOnChangeCallback(func(oldValue T, newValue T) {
+		f.logger.WithFields(logrus.Fields{
+			"old_value": oldValue,
+			"new_value": newValue,
+		}).Info("flag change detected")
+	})
+}
```

Comment:
Why not enable this by default? Having a history of the value is useful in debugging and I doubt people are going to change the value every second so it should not create a lot of noise?